projectview/src/main/java/org/jetbrains/bsp/bazel/projectview/parser/sections/ProjectViewSingletonSectionParser.java

```diff
@@ -0,0 +1,31 @@
+package org.jetbrains.bsp.bazel.projectview.parser.sections;
+
+import io.vavr.control.Try;
+import java.util.Optional;
+import org.jetbrains.bsp.bazel.projectview.model.sections.ProjectViewSingletonSection;
+import org.jetbrains.bsp.bazel.projectview.parser.splitter.ProjectViewRawSections;
+
+abstract class ProjectViewSingletonSectionParser<T extends ProjectViewSingletonSection>
+    extends ProjectViewSectionParser<Optional<T>> {
+
+  protected ProjectViewSingletonSectionParser(String sectionName) {
+    super(sectionName);
+  }
+
+  @Override
+  public Optional<T> parseOrDefault(ProjectViewRawSections rawSections, Optional<T> defaultValue) {
+    return parse(rawSections).or(() -> defaultValue);
+  }
+
+  @Override
+  public Optional<T> parse(ProjectViewRawSections rawSections) {
+    return rawSections.getLastSectionWithName(sectionName).map(this::parse).flatMap(Try::get);
+  }
+
+  @Override
+  protected Optional<T> parse(String sectionBody) {
+    return Optional.of(sectionBody.strip()).filter(body -> !body.isEmpty()).map(this::instanceOf);
+  }
+
+  protected abstract T instanceOf(String value);
```

Comment:
To avoid ambiguity with type-checking (like with Java's `instanceof` operator) I would name this method `instance` or `getInstance`.

Maybe there's a better name for the `value` argument as well, like `sectionBody`.

Reply:
> Maybe there's a better name for the value argument as well, like sectionBody.

well in this scope it is no longer a generic "section body" it is the specific value of a single value section

Reply:
wrt "value", the only code I could find that consumes `instanceOf` is:
```
  protected Optional<T> parse(String sectionBody) {
    return Optional.of(sectionBody.strip()).filter(body -> !body.isEmpty()).map(this::instanceOf);
  }
```
I suppose there's a fine line between "section" and "value"?

Reply:
section contains value (or list of values) and a name